models/yiiModels/YiiEventModel.php

```diff
@@ -0,0 +1,159 @@
+    
+    /**
+     * @var array
+     */
+    public $concernedItems; 
+    const CONCERNED_ITEMS = "concernedItems";
+    
+    /**
+     * @example 2019-01-02T00:00:00+01:00
+     * @var string 
+     */
+    public $date;
+    const DATE = "date";
+    
+    /**
+     * Specific properties
+     * @var array 
+     */
+    public $properties;
+    const PROPERTIES = "properties";
+    const RELATION = "relation";
+    const VALUE = "value";
+    
+    public function __construct($pageSize = null, $page = null) {
+        $this->wsModel = new WSEventModel();
+        ($pageSize !== null || $pageSize !== "") ? $this->pageSize = $pageSize 
+                : $this->pageSize = null;
+        ($page !== null || $page !== "") ? $this->page = $page : $this->page = null;
+    }
+    
+    /**
+     * @return array the rules of the attributes
+     */
+    public function rules() {
+       return [ 
+           [[YiiEventModel::URI], 'required'],
+           [[
+                YiiEventModel::TYPE, 
+                YiiEventModel::CONCERNED_ITEMS, 
+                YiiEventModel::DATE, 
+                YiiEventModel::PROPERTIES
+            ] , 'safe']
+        ]; 
+    }
+    
+    /**
+     * @return array the labels of the attributes
+     */
+    public function attributeLabels() {
+        return [
+            YiiEventModel::URI => 'URI', 
+            YiiEventModel::TYPE => Yii::t('app', 'Type'), 
+            YiiEventModel::PROPERTIES => Yii::t('app', 'Properties'), 
+            YiiEventModel::DATE => Yii::t('app', 'Date')
+        ];
+    }
+    
+    /**
+     * Allows to fill the attributes with the informations in the array given 
+     * @param array $array array key => value which contains the metadata of 
+     * an event
+     */
+    protected function arrayToAttributes($array) {
+        $this->uri = $array[YiiEventModel::URI];
+        $this->type = $array[YiiEventModel::TYPE];
+        if ($array[YiiEventModel::CONCERNED_ITEMS]) {
+            $this->concernedItems = get_object_vars($array[YiiEventModel::CONCERNED_ITEMS]);
+        } 
+        $this->date = $array[YiiEventModel::DATE];
+        $this->properties = $array[YiiEventModel::PROPERTIES];
+    }
+    
+    /**
+     * Allows to fill the property attribute with the information of the given 
+     * array
+     * @param array $array array key => value with the properties of an event 
+     */
+    protected function propertiesArrayToAttributes($array) {
+        if ($array[YiiEventModel::PROPERTIES] !== null) {
+            foreach ($array[YiiEventModel::PROPERTIES] as $property) {
+                $propertyToAdd = null;
+                $propertyToAdd[YiiEventModel::RELATION] = $property->relation; 
+                $propertyToAdd[YiiEventModel::VALUE] = $property->value;
+                $propertyToAdd[YiiEventModel::TYPE] = $property->type;
+                $this->properties[] = $property;
+            }
+        }
+    }
+
+    /**
+     * TODO not used yet
```

Comment:
Use the SILEX:todo tag. 
Why is it not used yet ? You should explain it in the comment